storm-core/src/jvm/org/apache/storm/metrics2/StormMetricRegistry.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.metrics2;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import org.apache.storm.Config;
+import org.apache.storm.cluster.DaemonType;
+import org.apache.storm.metrics2.reporters.StormReporter;
+import org.apache.storm.task.WorkerTopologyContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+
+public class StormMetricRegistry {
+
+    private static final Logger LOG = LoggerFactory.getLogger(StormMetricRegistry.class);
+
+    private static final MetricRegistry REGISTRY = new MetricRegistry();
+
+    private static final List<StormReporter> REPORTERS = new ArrayList<>();
+
+    private static String hostName = null;
+
+    public static <T> SimpleGauge<T>  gauge(T initialValue, String name, String topologyId, Integer port){
+        SimpleGauge<T> gauge = new SimpleGauge<>(initialValue);
+        String metricName = String.format("storm.worker.%s.%s-%s", topologyId, port, name);
```

Comment:
I am not sure that the naming convention takes into account all of the metadata that we want.

There are no components (bolt/spout names) so there is no way to know who acked, just that acks happened, possibly within a specific worker.  There is no host name or supervisor id.  I suppose most systems will add it in, so it is probably fine, but it would be good to call it out so reporters know they need to include it.

There is no owner.  Perhaps we don't care about the owner, but I would like to be able to sum up resource usage by owner easily in the future.

It would also be nice to have a way to parse these things back out of the name.  Some systems allow for reporting metadata with the metrics and I would prefer to be able to get back to the original metric name and report the rest as actual metadata.